### PR TITLE
docs: add tag/source/service details in datadog_logs sink

### PR DESCRIPTION
**Pull request by prognant**
Small change to expose tag/service/source field, it mainly targets user that would use the datadog_logs sinks to emit log that does not originate from the datadog_logs source.

### CODE REVIEW

File: docs/reference/components/sinks/datadog_logs.cue [hunk 1]
```diff
@@ -3,6 +3,13 @@ package metadata
 components: sinks: datadog_logs: {
 	title: "Datadog Logs"
 
+	description: """
+		Sends data to Datadog. The Datadog platform can leverage specifics fields (namely `ddtags`, `ddsource` and `service`) to further categorize
+		and filter logs. For further details on the purpose of those fields please check the [official Datadog documentation](\urls.datadog_tags).
+		The source (`ddsource` field) and service (`service` field) are expected to be plain text value. The list of tags (The `ddtags` field) should
+		be a single string of coma separated tags, e.g. `tag1:val1,tag2:val2,tag3:val3`.
```
Comment (binarylogic) [inline docs/reference/components/sinks/datadog_logs.cue:10]:
```suggestion
		Sends data to the [Datadog logs service](\(urls.datadog_logs)). The Datadog platform can leverage specifics fields (namely `ddtags`, `ddsource` and `service`) to further categorize
		and filter logs. For further details on the purpose of those fields please check the [official Datadog documentation](\(urls.datadog_tags)).
		The source (`ddsource` field) and service (`service` field) are expected to be plain text values. The list of tags (The `ddtags` field) should
		be a single string of coma separated tags, e.g. `tag1:val1,tag2:val2,tag3:val3`.
```

Comment (binarylogic) [inline docs/reference/components/sinks/datadog_logs.cue:10]: The last 2 sentences that describe the schema should probably be represented as structured data in the `input` field, agree? Currently the `input` field only accepts booleans on the types, but we should expand that to specify special fields like these.

cc @lucperkins and @jszwedko for thoughts.

Comment (prognant) [reply docs/reference/components/sinks/datadog_logs.cue]: As we do have #MetricInput/#MetricOutput & #LogOutput (as per https://github.com/timberio/vector/blob/master/docs/reference/components.cue#L402-L424). I would say we could imagine having #LogInput (or rename & reuse #LogOutput if there is no difference) to describe usable fields for sinks that can more or less strict requirement (e.g. https://github.com/timberio/vector/pull/7678/files#diff-90d1289b0a2da6889d20d7d009dcefcac4cf814c42158aa6ffafaa48d8e4b6d4R159-R172 would benefit from precise fields documentation). WDYT ? cc @lucperkins @jszwedko

Comment (jszwedko) [reply docs/reference/components/sinks/datadog_logs.cue]: 👍 that seems reasonable. It seems like we'd just add a `fields: #Schema` to `#LogInput`. We would need to make sure we still model the fact that events can have arbitrary fields outside of the special ones.

File: docs/reference/components/sinks/datadog_logs.cue [hunk 2]
```diff
@@ -3,6 +3,13 @@ package metadata
 components: sinks: datadog_logs: {
 	title: "Datadog Logs"
 
+	description: """
+		Sends data to the [Datadog logs service](\(urls.datadog_logs)). The Datadog platform can leverage specifics fields (namely `ddtags`, `ddsource` and `service`) to further categorize
```
Comment (lucperkins) [inline docs/reference/components/sinks/datadog_logs.cue:7]: s/the Datadog logs service/Datadog Log Management

s/leverage/use

s/specifics/specific

comma after `ddsource`

File: docs/reference/components/sinks/datadog_logs.cue [hunk 3]
```diff
@@ -3,6 +3,13 @@ package metadata
...
+		and filter logs. For further details on the purpose of those fields please check the [official Datadog documentation](\(urls.datadog_tags)).
```
Comment (lucperkins) [inline docs/reference/components/sinks/datadog_logs.cue:8]: For further details on the purpose of those fields, refer to the official Datadog documentation

File: docs/reference/components/sinks/datadog_logs.cue [hunk 4]
```diff
@@ -3,6 +3,13 @@ package metadata
...
+		The source (`ddsource` field) and service (`service` field) are expected to be plain text values. The list of tags (The `ddtags` field) should
```
Comment (lucperkins) [inline docs/reference/components/sinks/datadog_logs.cue:9]: s/plain text/plaintext

s/The `ddtags` field/the `ddtags` field

File: docs/reference/components/sinks/datadog_logs.cue [hunk 5]
```diff
@@ -3,6 +3,13 @@ package metadata
...
+		be a single string of coma separated tags, e.g. `tag1:val1,tag2:val2,tag3:val3`.
```
Comment (lucperkins) [inline docs/reference/components/sinks/datadog_logs.cue:10]: comma-separated

**Review by lucperkins — CHANGES_REQUESTED** [review]

**Comment by binarylogic** [comment]
@prognant should we close this? In general, I think this description should go under the `how_it_works` key.

**Comment by prognant** [comment]
> @prognant should we close this? In general, I think this description should go under the `how_it_works` key.

@binarylogic closing it, the material written here is mostly integrated in datadog upcoming docs update.

**Pull request closed without merging**